kernel/src/syscall/sched_getparam.rs

```diff
@@ -4,16 +4,18 @@ use super::{sched_getattr::access_sched_attr_with, SyscallReturn};
 use crate::{prelude::*, sched::SchedPolicy, thread::Tid};
 
 pub fn sys_sched_getparam(tid: Tid, addr: Vaddr, ctx: &Context) -> Result<SyscallReturn> {
+    if addr == 0 {
+        return_errno_with_message!(Errno::EINVAL, "invalid user space address");
+    }
+
     let policy = access_sched_attr_with(tid, ctx, |attr| Ok(attr.policy()))?;
     let rt_prio = match policy {
         SchedPolicy::RealTime { rt_prio, .. } => rt_prio.get().into(),
         _ => 0,
     };
 
-    let space = ctx.user_space();
-    space
-        .write_val(addr, &rt_prio)
-        .map_err(|_| Error::new(Errno::EINVAL))?;
+    let user_space = ctx.user_space();
+    user_space.write_val(addr, &rt_prio)?;
```

Comment:
Simplified as `ctx.user_space().write_val(addr, &rt_prio)?` since user_space is not used anywhere else?